libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -311,7 +311,7 @@ void ContractCompiler::appendCalldataUnpacker(TypePointers const& _typeParameter
 	{
 		// stack: v1 v2 ... v(k-1) base_offset current_offset
 		TypePointer type = parameterType->decodingType();
-		solAssert(type, "No decoding type found.");
+		solUnimplementedAssert(type, "No decoding type found.");
```

Comment:
Perhaps take this with the other change too (in general I like those specific `solUnimplementedAssert`s better from a user's perspective).